File: November_Leetcode_Challenge_2022/Day1/1706._Where_Will _the_Ball_Fall.java
class Solution {
    public int[] findBall(int[][] grid) {
        int col=grid[0].length;
        int []ans= new int[col];
        for(int j=0;j<col;j++){
            ans[j]=dfs(0,j,grid);
        }
        return ans;
    }
    private int dfs(int i,int j,int [][]grid){
        if(i>=grid.length){
            return j;
        }
        if(grid[i][j]==1 && j+1<grid[0].length && grid[i][j+1]==1){
            return dfs(i+1,j+1,grid);
        }
        else if(grid[i][j]==-1 && j-1>=0 && grid[i][j-1]==-1){
            return dfs(i+1,j-1,grid);
        }
        else if(grid[i][j]==1 && j+1>=grid[0].length){
            return -1;
        }
        else{
            return -1;
        }
       // return -1;
    }
}

File: November_Leetcode_Challenge_2022/Day1/1706._Where_Will _the_Ball_Fall.cpp
class Solution {
public:
    int dfs(int i, int j, vector<vector<int>>& grid )
    {
        if(i>= grid.size())
        {
            return j;
        }
        //left to right 
        if(grid[i][j] == 1 && j+1<grid[0].size() && grid[i][j+1] == 1)
        {
            return dfs(i+1,j+1,grid);
        }
        else if(grid[i][j] == -1 && j-1>=0 && grid[i][j-1] == -1)
        {
            return dfs(i+1,j-1,grid);
        }
        else if(grid[i][j]==1 && j+1>=grid[0].size() )
        {
            return -1;
        }
        else
        {
            return -1;
        }
        
    }
    vector<int> findBall(vector<vector<int>>& grid) {
      int row = grid.size();
      int col = grid[0].size();
      vector<int>ans(col);
      for(int j=0;j<col;j++)
      {
          ans[j]= dfs(0,j,grid);
      }
        return ans;
    }
};
